src/Umbraco.Web/Editors/CodeFileController.cs

```diff
@@ -0,0 +1,155 @@
+﻿using AutoMapper;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web.Http;
+using Umbraco.Core.Models;
+using Umbraco.Web.Models.ContentEditing;
+using Umbraco.Web.Mvc;
+using Umbraco.Web.WebApi;
+
+namespace Umbraco.Web.Editors
+{
+    [PluginController("UmbracoApi")]
+    public class CodeFileController : UmbracoAuthorizedJsonController
```

Comment:
There is no authorization on this which means any authenticated user can access all parts of this controller. If you want to lock a controller or actions down based on what tree the user has access to you can use something like (for example)

```
[UmbracoTreeAuthorize(Constants.Trees.DataTypes, Constants.Trees.DocumentTypes, Constants.Trees.MediaTypes, Constants.Trees.MemberTypes)]
```

However, because we don't have true permissions per tree, all this does is check if the user has access to the tree type based on what section they have access to (something I'd like to fix sometime so we have true permissions per tree).